config/opensearch_dashboards.yml

```diff
@@ -281,6 +281,14 @@
 #    'ff00::/8',
 # ]
 
+# Optional setting that enables you to specify a path to PEM files for the certificate
+# authority for your connected datasources.
+#data_source.ssl.certificateAuthorities: [ "/path/to/your/CA.pem" ]
```

Comment:
@BionIT @ZilongX This configuration introduces a potential security risk by implying a global setting for certificate authorities across all data sources. This approach might not meet sysadmins' expectations, who typically require the ability to specify distinct certificate authorities for each data source for enhanced security. Could we consider revising this design to allow each data source to be securely and separately configured, ensuring attributes such as the certificate authority are isolated?

Reply:
@peternied I have called that out on the corresponding issue here: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4524#issuecomment-1989323830

These settings would explicitly be for a common config if TLS is common between the attached data sources. TLS per data source would require UX when configuring a data source.

Reply:
@peternied Do you think that this should block the PR or can we add this shared setting to unblock testing with HTTPS in the security-dashboards-plugin? I can file an issue around configured TLS per data-source to track the work. That would either require UX where TLS is configured when adding a data source or possibly a separate config file from `opensearch_dashboards.yml` where you can config your data sources from yml instead of through OSD pages.

Reply:
@BionIT @ZilongX Is there a plan to configure multiple data-sources through a yml file?

Reply:
Filed an issue to track datasource-specific TLS configuration: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6226